internal/pkg/cli/app_show.go

```diff
@@ -0,0 +1,312 @@
+			if err != nil {
+				return fmt.Errorf("connect to environment datastore: %w", err)
+			}
+			opts.projectLister = ssmStore
+			opts.projectGetter = ssmStore
+			opts.appLister = ssmStore
+			opts.appGetter = ssmStore
+			opts.envLister = ssmStore
+
+			return nil
+		}),
+		RunE: runCmdE(func(cmd *cobra.Command, args []string) error {
+			if err := opts.Ask(); err != nil {
+				return err
+			}
+			identifier, err := describe.NewWebAppDescriber(opts.ProjectName(), opts.appName)
+			if err != nil {
+				return fmt.Errorf("creating identifier for application %s in project %s: %w", opts.appName, opts.ProjectName(), err)
+			}
+			opts.identifier = identifier
+			if err := opts.Execute(); err != nil {
+				return err
+			}
+
+			return nil
+		}),
+	}
+	// The flags bound by viper are available to all sub-commands through viper.GetString({flagName})
+	cmd.Flags().StringVarP(&opts.appName, appFlag, appFlagShort, "", appFlagDescription)
+	cmd.Flags().BoolVar(&opts.ShouldOutputJSON, jsonFlag, false, jsonFlagDescription)
+	cmd.PersistentFlags().StringP(projectFlag, projectFlagShort, "" /* default */, projectFlagDescription)
+	viper.BindPFlag(projectFlag, cmd.PersistentFlags().Lookup(projectFlag))
+	return cmd
+}
+
+// Ask asks for fields that are required but not passed in.
+func (opts *ShowAppOpts) Ask() error {
+	// get project
+	if opts.ProjectName() == "" {
+		projectName, err := opts.selectProject()
+		if err != nil {
+			return err
+		}
+		opts.projectName = projectName
+	} else {
+		msg := fmt.Sprintf(
+			"Looks like you are using a workspace that's registered to project %s.\nWe'll use that as your project.",
+			color.HighlightResource(opts.ProjectName()))
+		log.Infoln(msg)
+	}
+
+	// get application of the project
+	if opts.appName == "" {
+		appName, err := opts.selectApplication()
+		if err != nil {
+			return err
+		}
+		opts.appName = appName
+	}
+
+	return nil
+}
+
+// Execute shows the applications through the prompt.
+func (opts *ShowAppOpts) Execute() error {
+	proj, err := opts.projectGetter.GetProject(opts.ProjectName())
+	if err != nil {
+		return fmt.Errorf("getting project: %w", err)
+	}
+	opts.proj = proj
+
+	app, err := opts.appGetter.GetApplication(opts.ProjectName(), opts.appName)
+	if err != nil {
+		return fmt.Errorf("getting application: %w", err)
+	}
+	opts.app = app
+
+	envs, err := opts.envLister.ListEnvironments(opts.ProjectName())
+	if err != nil {
+		return fmt.Errorf("listing environments: %w", err)
+	}
+	opts.envs = envs
+
+	uris := make([]*describe.WebAppURI, len(opts.envs))
+	for ind, env := range envs {
+		webAppURI, err := opts.identifier.URI(env.Name)
+		if err != nil {
+			log.Infoln(fmt.Sprintf(`Could not retrieve the URI from environment %s.
+Please try to deploy %s with environment %s first before getting the URI.`, color.HighlightUserInput(env.Name), color.HighlightUserInput(opts.appName), color.HighlightUserInput(env.Name)))
+			uris[ind] = &describe.WebAppURI{
+				DNSName: "",
+				Path:    "",
+			}
```

Comment:
Is there a way of checking if the error is because the application does not exist in the environment? 

I think the ideal UX would be to skip writing the row if the environment is not deployed.